### ISSUE TITLE
Selecting from a view shows column names as all lowercase

### ISSUE THREAD

**Issue by 0x4261756D** [dolthub/dolt#6683]
The column names of a view should be returned case-sensitive when selecting from them.
When describing the table this works correctly, when selecting it does not.
For normal tables the case is also preserved in both cases.

**Steps to reproduce (slightly modified version of https://docs.dolthub.com/concepts/dolt/sql/views#example):**
In an empty repo:
```sql
create table salaries (nAmE varchar(255), sAlArY int, primary key(nAmE));
insert into salaries values ('Jim', 120000), ('Bob', 240000), ('Sally', 360000);
create view monthly_salaries as select name as CaSeSeNsItIvE, salary/12 as mOnThLy_PaY from salaries;
```
**Output:**
```sql
describe salaries;
```
```
+--------+--------------+------+-----+---------+-------+
| Field  | Type         | Null | Key | Default | Extra |
+--------+--------------+------+-----+---------+-------+
| nAmE   | varchar(255) | NO   | PRI | NULL    |       |
| sAlArY | int          | YES  |     | NULL    |       |
+--------+--------------+------+-----+---------+-------+
```
```sql
select * from salaries;
```
```
+-------+--------+
| nAmE  | sAlArY |
+-------+--------+
| Bob   | 240000 |
| Jim   | 120000 |
| Sally | 360000 |
+-------+--------+
```

```sql
describe monthly_salaries;
```
```
+---------------+-------------+
| casesensitive | monthly_pay |
+---------------+-------------+
| Bob           | 20000.0000  |
| Jim           | 10000.0000  |
| Sally         | 30000.0000  |
+---------------+-------------+
```
```sql
select * from monthly_salaries;
```
```
+---------------+--------------+------+-----+---------+-------+
| Field         | Type         | Null | Key | Default | Extra |
+---------------+--------------+------+-----+---------+-------+
| CaSeSeNsItIvE | varchar(255) | NO   |     | NULL    |       |
| mOnThLy_PaY   | decimal(2,0) | YES  |     | NULL    |       |
+---------------+--------------+------+-----+---------+-------+
```

**Comment by fulghum** [comment]
Hey @0x4261756D, thanks for taking the time to report this one! I was able to easily repro this and just opened a PR with a fix. I'll run the Dolt CI test suite on it next and we'll get this fix checked in for you asap.

**Comment by fulghum** [comment]
This fix is now released in [Dolt version 1.16.1](https://github.com/dolthub/dolt/releases/tag/v1.16.1).

Thanks again for taking the time to let us know about this one! 🙏 Please let us know if you run into any other issues and we'll be happy to investigate!
